sunpy/net/dataretriever/attrs/__init__.py

```diff
@@ -1,3 +1,8 @@
 from . import goes
+from . import adapt
 
-__all__ = ['goes']
+__all__ = ['goes', 'adapt']
+
+# Define a custom __dir__ to restrict tab-completion to __all__
+def __dir__():
+    return __all__
```

Comment:
I do not think we need this for this file since the only two items in this file are covered by all.